tests/rl/test_hooks.py

```diff
@@ -0,0 +1,472 @@
+
+        # Should not run at step 5
+        context = self._create_context(step=5)
+        self.assertFalse(hook.should_run(context))
+
+        # Should run at step 10
+        context = self._create_context(step=10)
+        self.assertTrue(hook.should_run(context))
+
+        # Should not run at step 15
+        context = self._create_context(step=15)
+        self.assertFalse(hook.should_run(context))
+
+        # Should run at step 20
+        context = self._create_context(step=20)
+        self.assertTrue(hook.should_run(context))
+
+    def test_periodic_hook_start_step(self):
+        """Test that periodic hooks respect the start_step parameter."""
+
+        class TestHook(PeriodicHook):
+            def run(self, context):
+                return {"ran": True}
+
+        hook = TestHook(frequency=10, start_step=50)
+
+        # Should not run before start_step
+        context = self._create_context(step=10)
+        self.assertFalse(hook.should_run(context))
+
+        context = self._create_context(step=50)
+        self.assertFalse(hook.should_run(context))
+
+        # Should run at step 60 (first multiple of 10 after start_step)
+        context = self._create_context(step=60)
+        self.assertTrue(hook.should_run(context))
+
+
+class TestEvaluationHook(unittest.TestCase):
+    """Test evaluation hook functionality."""
+
+    def setUp(self):
+        """Set up test fixtures."""
+        self.mock_worker = MagicMock()
+        self.mock_curriculum_actor = MagicMock()
+        self.rng = jrandom.PRNGKey(42)
+
+    def _create_context(self, **kwargs):
+        """Helper to create HookContext with required fields."""
+        defaults = {
+            "worker": self.mock_worker,
+            "step": 100,
+            "rng": self.rng,
+            "curriculum_actor": self.mock_curriculum_actor,
+        }
+        defaults.update(kwargs)
+        return HookContext(**defaults)
+
+    def test_evaluate_lesson_hook(self):
+        """Test lesson evaluation hook configuration."""
+        from marin.rl.types import RolloutBatch, RolloutGroup, Rollout, RolloutMetadata
+        import numpy as np
+
+        hook = EvaluateLessonHook(frequency=10, start_step=0, n_examples=4)
+
+        # Create a mock batch
+        rollout = Rollout(
+            env_name="test",
+            env_example_id="test_example",
+            prompt_tokens=np.array([1, 2, 3], dtype=np.int32),
+            response_tokens=np.array([4, 5, 6], dtype=np.int32),
+            response_logprobs=np.array([0.1, 0.2, 0.3], dtype=np.float32),
+            token_rewards=np.array([0.5, 0.6, 0.7], dtype=np.float32),
+            episode_reward=1.0,
+        )
+        batch = RolloutBatch(
+            groups=[RolloutGroup(rollouts=[rollout])],
+            metadata=RolloutMetadata(worker_id="test", timestamp=0.0, weight_step=0),
+        )
+
+        self.mock_worker._sample_batch = MagicMock(return_value=(batch, None))
```

Comment:
this is a lot of mocks, prefer to define a MockWorker and reuse.

it doesn't seem like this is exercising functionality we really care about though. you already test the "hook is called at step X" in your periodic hook tests. for the evaluation, we can either test that it actually evaluates something we care about, or remove these and rely on our integration tests. testing that it calls an internal function is again fragile.